client/src/main/java/io/pravega/client/stream/impl/EventPointerImpl.java

```diff
@@ -52,4 +61,62 @@ int getEventLength() {
     public EventPointerInternal asImpl() {
         return this;
     }
+    
+    public static EventPointer fromString(String eventPointer) {
+        int i = eventPointer.lastIndexOf(":");
```

Comment:
Would a regex help here? Based on the input, this could throw IndexOutOfBounds or other kinds of weird exceptions, not just IllegalArgumentExceptions.

Reply:
I didn't see a clean way. If you do could you post it here?